projects/storefrontlib/src/cms-components/order-confirmation/components/order-confirmation-thank-you-message/order-confirmation-thank-you-message.component.ts

```diff
@@ -15,14 +29,78 @@ import { Observable } from 'rxjs';
 export class OrderConfirmationThankYouMessageComponent
   implements OnInit, OnDestroy {
   order$: Observable<Order>;
+  isGuestCustomer = false;
+  orderGuid: string;
+
+  subscription: Subscription;
+
+  guestRegisterForm: FormGroup = this.fb.group(
+    {
+      password: [
+        '',
+        [Validators.required, CustomFormValidators.passwordValidator],
+      ],
+      passwordconf: ['', Validators.required],
+    },
+    { validator: this.matchPassword }
+  );
 
-  constructor(protected checkoutService: CheckoutService) {}
+  constructor(
+    checkoutService: CheckoutService,
+    userService: UserService, // tslint:disable-line
+    routingService: RoutingService, // tslint:disable-line
+    authService: AuthService, // tslint:disable-line
+    fb: FormBuilder // tslint:disable-line
+  );
+  /**
+   * @deprecated since version 1.x
+   *
+   *  TODO(issue:#1178) deprecated since 1.x
+   */
+  constructor(checkoutService: CheckoutService);
+  constructor(
+    protected checkoutService: CheckoutService,
+    protected userService?: UserService,
+    protected routingService?: RoutingService,
+    protected authService?: AuthService,
+    protected fb?: FormBuilder
+  ) {}
 
   ngOnInit() {
-    this.order$ = this.checkoutService.getOrderDetails();
+    this.order$ = this.checkoutService.getOrderDetails().pipe(
+      tap(order => {
+        this.isGuestCustomer = order.guestCustomer;
+        this.orderGuid = order.guid;
+      })
+    );
   }
 
   ngOnDestroy() {
     this.checkoutService.clearCheckoutData();
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  submit() {
+    if (this.isGuestCustomer) {
+      this.userService.registerGuest(
+        this.orderGuid,
+        this.guestRegisterForm.value.password
+      );
+      if (!this.subscription) {
+        this.subscription = this.authService.getUserToken().subscribe(token => {
+          if (token.access_token) {
+            this.routingService.go({ cxRoute: 'home' });
+          }
+        });
+      }
+    }
+  }
+
+  private matchPassword(ac: AbstractControl): { NotEqual: boolean } {
```

Comment:
Since this is now used in two places should we add it to the `custom-form-validators.ts`?